File: 0035-search-insert-position/0035-search-insert-position.js
/**
 * @param {number[]} nums
 * @param {number} target
 * @return {number}
 */
var searchInsert = function(nums, target) {
    var low = -1;
    var high = nums.length;

    var mid;
    while (low < high-1) {
        // console.log(`low: ${low}, high: ${high}`)
        mid = Math.floor((low+high)/2);
        // console.log(`checking index: ${mid}, val: ${nums[mid]}, target: ${target}`)
        if (nums[mid] === target) {
            // console.log('equal');
            return mid;
        } else if (nums[mid] > target) {
            // console.log('greater');
            high = mid;
        } else if (nums[mid] < target) {
            // console.log('less than');
            low = mid;
        }
    }

    return high;
};

File: 0035-search-insert-position/0035-search-insert-position.java
class Solution {
    public int searchInsert(int[] nums, int target) {
        int lo = 0;
        int hi = nums.length;
        
        while (lo < hi) {
            int mid = (int) Math.floor((lo+hi)/2);
            
            if (nums[mid] == target) {
                return mid;
            } else if (nums[mid] > target) {
                hi = mid;
            } else {
                lo = mid + 1;
            }
        }
        
        return lo;
    }
}